### PR TITLE
Shelley testing

### PR DESCRIPTION
**Pull request by nc6**
- Some infra restructuring in preparation for more tests
- Test with multiple values of the decentralisation parameter.

### CODE REVIEW

File: ouroboros-consensus-shelley/test/Test/ThreadNet/RealTPraos.hs [hunk 1]
```diff
@@ -126,6 +123,10 @@ prop_simple_real_tpraos_convergence k d
     epochSize :: EpochSize
     epochSize = sgEpochLength genesisConfig
 
+-------------------------------------------------------------------------
+-- Test config
+-------------------------------------------------------------------------
+
```
Comment (mrBliss) [inline ouroboros-consensus-shelley/test/Test/ThreadNet/RealTPraos.hs]: Throughout consensus we use the following style:
```suggestion
{-------------------------------------------------------------------------------
  Test config
-------------------------------------------------------------------------------}
```

File: ouroboros-consensus-shelley/test/Test/ThreadNet/Infra/Shelley.hs [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+module Test.ThreadNet.Infra.Shelley
+  ( CoreNode(..)
+  , CoreNodeKeyInfo(..)
+  , genCoreNode
+  , mkGenesisConfig
+  , coreNodeKeys
+  , mkProtocolRealTPraos
+  , tpraosSlotLength
+  )
+where
```
Comment (mrBliss) [inline ouroboros-consensus-shelley/test/Test/ThreadNet/Infra/Shelley.hs:18]: Edsko-style indentation:

```suggestion
module Test.ThreadNet.Infra.Shelley (
    CoreNode(..)
  , CoreNodeKeyInfo(..)
  , genCoreNode
  , mkGenesisConfig
  , coreNodeKeys
  , mkProtocolRealTPraos
  , tpraosSlotLength
  ) where
```

**Comment by nc6** [comment]
bors merge

**Pull request merged**